Сетка страниц и крупных блоков на флексах

AlyonaPrikhodchenko opened the pull request:
---
:mortar_board: [Сетка страниц и крупных блоков на флексах](https://up.htmlacademy.ru/htmlcss/33/user/1875081/tasks/7)

:boom: https://htmlacademy-htmlcss.github.io/1875081-sedona-33/4/

AlyonaPrikhodchenko committed:
background в одном виде

Наставник, я исправилась))

AlyonaPrikhodchenko committed:
Сетки. Начало

AlyonaPrikhodchenko committed:
Cетки

AlyonaPrikhodchenko commented:
Привет! Я отправила дз на проверку, но чуть не доделала вторую страницу. На первой у меня проблема со списком и со списком в списке в плане построения сетки. Еще есть чутка проблемы с пиксель перфект, к примеру, у меня почему-то в селекте текст выходит больше, чем на макете. Просто до завтра не успеваю все сделать, так что пока как есть, потом подредачу.

И кстати я иконки типа фейсбук вставила свг в разметке, что-то у меня какая-то проблема с псевдоэлементами или как их правильно вставить. Потому прости,  я подглядела разметку на сайте академии и вот так вставила. Но если подскажешь, как сделать через css, сделаю так. А-то гляди на защите так не прокатит

AlyonaPrikhodchenko commented on catalog.html:
```diff
@@ -133,14 +137,15 @@ <h3 class="filters-header-title">Гостиницы Седоны</h3>
           <h2 class="visually-hidden">Список найденных гостиниц</h2>
           <header class="hotels-header">
             <h3 class="hotels-title">Найдено гостиниц: <span>38</span></h3>
             <select class="select-control" aria-label="Сортировка">
```
В этом селекте размер шрифта почему-то не совпадает с pp

AlyonaPrikhodchenko committed:
Доработка сеток

AlyonaPrikhodchenko committed:
Преимущества

Оформила список преимуществ

ad0v0 commented:
**Проверка  сетки страницы и крупных блоков на флексах:**

1) Проверка работы согласно критериям:
- ☑️ [TEST-03. Размеры страницы](https://up.htmlacademy.ru/htmlcss/32/criteries#TEST-03):
-- http://joxi.ru/nAy1J5bt93xvkA обнулить паддинги у документа страницы, чтобы не создавал горизонтального скроллбара на минимальной для макета ширины. Откуда берутся 1217 пых = 1200 пых сама контентная ширина + 17 пых вертикальный скроллбар на десктопе
- ☑️ [TEST-04. Переполнение](https://up.htmlacademy.ru/htmlcss/32/criteries#TEST-04):
-- http://joxi.ru/l2ZqxjlHlebGXm сохраняем отступы
-- http://joxi.ru/nAy1J5bt93xlEA позволить расти кнопкам в пагинации, чтобы числовое значение сохраняло пропорции с отступами и центрирование
- 🎌 [TEST-06. Pixel Perfect](https://up.htmlacademy.ru/htmlcss/32/criteries#TEST-06):
-- http://joxi.ru/Vm6Y4w9S37g8Rr поправить подвал
-- http://joxi.ru/zAN361ac1Kp0nr расположение фильтрации цены
-- http://joxi.ru/Dr8YPw4SKyG5Xm отступы у карточек
-- http://joxi.ru/krDQRKjIGBPo3m расположение блока с подпиской
-- http://joxi.ru/bmoKJN0COQpn7m вертикали тоже сохраняем отступы 
-- http://joxi.ru/V2V73MdsBvgXGA не используем фикс высоту, чтобы текст не выпадал из своего блока 
- ✅ [PROJ-04. Закомментированный код](https://up.htmlacademy.ru/htmlcss/32/criteries#PROJ-04) - ок
- ✅ [HTML-06. Лишние элементы](https://up.htmlacademy.ru/htmlcss/32/criteries#HTML-06) - ок 
2) Разное:
- ☑️ Стики футер:
-- http://joxi.ru/vAW8OGLt36vX9r реализовать sticky footer. Доп инфа, на которую можно опираться: https://css-tricks.com/couple-takes-sticky-footer/. Нас интересует вариант с флексами, но можно и гриды, остальное не рассматриваем
- ✅ Контейнер-центровщик - ок 
- ✅ Флекс свойства - ок
- ✅ Расположение блоков в потоке - ок 
- ➡️ http://joxi.ru/BA0YGw1S14y73r на всякий пожарный уточню, что по вертикали мы всегда выставляем значение на ноль, тк оно является единственным верным при проверке ПП 
- ☑️ http://joxi.ru/Vm6Y4w9S37gdRr поправить цвет фона документа страницы на f2f2f2 - текущей цвет это цвет полотна в фигме
- ☑️ http://joxi.ru/DrlWBybUK3WzQA можно оптимизировать подобное обнуление: создаем правило по селектору с классом reset-list и выносим все обнуление для списков в него, затем добавляем этот класс каждому списку в разметке
3) Общее впечатление от выполненной работы: кайфанула от от визуала как только его увидела ❤️ Все четенько и на своих местах 👍🏻 Код тоже пушка: реализовано по принципу минимализма - ничего лишнего, только то, что нужно использовано 👌🏻 

🎌 - есть замечание, надо править
➡️ - есть комментарий, желательно учесть
✅ - выполнено 
☑️ - правка по замечанию внесена
❓ - есть вопрос, желательно ответить

ПС Ошибки , которые дублируются в коде, в замечаниях не дублирую, но перепроверять и править нужно везде и обязательно

AlyonaPrikhodchenko committed:
Доработки по сетке

Исправила переполнение, получше подогнала под pp

AlyonaPrikhodchenko committed:
Добавила гриды

Список преимуществ, каталог, форму поиска и кнопки изменения каталога сделала гридами, вместо флексов.

ad0v0 commented:
- http://joxi.ru/MAj5BkbSdE5j82 текст в select - размер шрифта уменьшить до 16 пых
- Здесь нужны уточнения:
> На первой у меня проблема со списком и со списком в списке в плане построения сетки

- а можешь плс на примере одной иконки показать, как ты пыталась реализовать через псевдоэлемент:
> И кстати я иконки типа фейсбук вставила свг в разметке, что-то у меня какая-то проблема с псевдоэлементами или как их правильно вставить.

- http://joxi.ru/MAj5BkbSdE5x82 сделать тень с прозрачностью - см в фигме
